mantidproject/mantid
# Muon Background Corrections - Perform fit to find background

**Description of work.**
This PR ensures the background of a fit is now calculated and populated into the data table. The background is not yet subtracted from any data - this will be done in a separate PR.

**To test:**
1. Comment out this line https://github.com/mantidproject/mantid/blob/4e3edf60bb556bdf1de046309270711e1ce1fc33/scripts/Muon/GUI/Common/corrections_tab_widget/background_corrections_view.py#L55
2. Open Muon Analysis and load PSI data
3. Go Corrections tab. Select `Auto` background corrections
4. Try the following combinations:
 - `Auto` with `Flat Background` between `5.0` and `9.8` should show `Correction skipped`
 - `Auto` with `Flat Background + Exp Decay` between `5.0` and `9.8` should show `Correction success`
 - `Auto` with `Flat Background` between `0` and `0.14` should show `Correction success` for timezero set to 0.0

5. Try unticking `Apply parameter changes to all domains`. Then change the start or end X of a row. A fit will be performed on the single row and update the `A0` in the table.




Part of #31816

*This does not require release notes* because **it is unfinished and still in development**.

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
